backend/utils.py

```
@@ -0,0 +1,343 @@
+        # Generate multimodal embedding
+        result = vo.multimodal_embed(inputs, model="voyage-multimodal-3")
+        
+        # Return the embedding vector
+        return result.embeddings[0]
+        
+    except Exception as e:
+        print(f"Voyage AI embedding error: {e}")
+        # Return mock embedding on error
+        return [0.1] * 1024  # Voyage embeddings are typically 1024-dimensional
+
+def extract_diy_metadata(vision_data: dict, name: str, category: str):
```

Ruff (0.12.2)

Unused function argument: `name`

(ARG001)